client/driver/docker.go

```diff
@@ -562,22 +562,22 @@ func (d *DockerDriver) HealthCheck(req *cstructs.HealthCheckRequest, resp *cstru
 	client, _, err := d.dockerClients()
 	if err != nil {
 		d.logger.Printf("[WARN] driver.docker: failed to retrieve Docker client in the process of a docker health check: %v", err)
-		dinfo.HealthDescription = fmt.Sprintf("Failed retrieving Docker client: %v", err)
+		dinfo.HealthDescription = fmt.Sprintf("Docker driver failed retrieving client: %v", err)
```

Comment:
Lets remove the name from the message and instead populate the details map here (https://github.com/hashicorp/nomad/blob/master/client/client.go) to include the driver name. Then on the CLI outputting of the node events lets detect Subsystem Driver and change the output to be `Driver: %s", Details["driver"]`. This way we can always emit the driver name in a clean fashion.